src/fastertransformer/layers/attention_layers/LlamaDecoderSelfAttentionLayer.h

```diff
@@ -73,6 +74,25 @@ class LlamaDecoderSelfAttentionLayer: public BaseAttentionLayer<T> {
     const int int8_mode_ = 0;
 
 public:
+    LlamaDecoderSelfAttentionLayer(size_t           max_batch_size,
```

Comment:
instead of adding new constructors, can we modify existing ones by adding `rope_scaling_factor` param?

Reply:
I modified the existing constructors to account for it. I wanted to keep it safe in case any other dependencies.

Reply:
The constructor at line 96, was it implemented in the cc file?